hq_superset/__init__.py

```diff
@@ -8,4 +8,11 @@ def patch_superset_config(config):
 def flask_app_mutator(app):
     # Import the views (which assumes the app is initialized) here
     # return
-    pass
+    from . import hq_domain
+    from . import views
+    from superset.extensions import appbuilder
+    appbuilder.add_view(views.SelectDomainView, 'Select Domain')
+    app.before_request_funcs.setdefault(None, []).append(
```

Comment:
I think you should be using the [`@before_request`](https://flask.palletsprojects.com/en/2.1.x/api/#flask.Flask.before_request) decorator rather than mutating this list here: 
https://flask.palletsprojects.com/en/2.1.x/api/#flask.Flask.before_request_funcs

>To register a function, use the [before_request()](https://flask.palletsprojects.com/en/2.1.x/api/#flask.Flask.before_request) decorator.
> 
> This data structure is internal. It should not be modified directly and its format may change at any time.

Reply:
Using the decorator doesn't work in the case of superset/FAB, because the way the superset makes the app object available. The app object needs to be available when using it as decorator. That's the reason, I had to do it this way.

Reply:
Instead of this line, can you just do this? 

```suggestion
    app.before_request(hq_domain.before_request_hook)
```

Reply:
I think this means that any other hooks elsewhere will get removed, no?

Reply:
I don't think so. This would be the same as decorating the function directly. Using `before_request` will do exactly what you have done here, but in the "supported way".

https://github.com/pallets/flask/blob/bd56d19b167822a9a23e2e9e2a07ccccc36baa8d/src/flask/scaffold.py#L555-L556